test_util/marathon.py

```diff
@@ -202,6 +202,82 @@ def _pool_for_marathon_app(app_id):
             raise Exception("Application deployment failed - operation was not "
                             "completed in {} seconds.".format(timeout))
 
+    def deploy_marathon_pod(self, pod_definition, timeout=300, check_health=False):
+        """Deploy a pod to marathon
+
+        This function deploys an a pod and then waits for marathon to
+        aknowledge it's successfull creation or fails the test.
+
+        It waits until all the instances reach tasksRunning and then tasksHealthy state.
+
+        Args:
+            pod_definition: a dict with pod definition as specified in
+                            Marathon API
+            timeout: a time to wait for the pod to reach 'Healthy' status
+                     after which the test should be failed.
+            check_health: wait until Marathon reports tasks as healthy before
+                          returning
+
+        Returns:
+            Scaling instance count
+        """
+        r = self.post('/pods', json=pod_definition)
+        logging.info('Response from marathon: {}'.format(repr(r.json())))
+        assert r.ok
+
+        @retrying.retry(wait_fixed=1000, stop_max_delay=timeout * 1000,
+                        retry_on_result=lambda ret: ret is None,
```

Comment:
retrying on return None feels a little awkward to me. Can we do retry on False instead? Otherwise, it seems like you may expect return data["scaling"]["instances"] to be None in some cases.